pyeudiw/satosa/backend.py

```diff
@@ -0,0 +1,195 @@
+        self.redirect_url = config['redirect_endpoint']
+        self.request_url = config['request_endpoint']
+        self.error_url = config['error_url']
+
+        self.client_id = config['wallet_relying_party']['client_id']
+        self.complete_redirect_url = config['wallet_relying_party']['redirect_uris'][0]
+        self.complete_request_url = config['wallet_relying_party']['request_uris'][0]
+
+        self.qr_settings = config['qr_code_settings']
+        self.config = config
+
+    def register_endpoints(self):
+        """
+        Creates a list of all the endpoints this backend module needs to listen to. In this case
+        it's the authentication response from the underlying OP that is redirected from the OP to
+        the proxy.
+        :rtype: Sequence[(str, Callable[[satosa.context.Context], satosa.response.Response]]
+        :return: A list that can be used to map the request to SATOSA to this endpoint.
+        """
+        url_map = []
+        url_map.append(
+            (f"^{self.entity_configuration_url.lstrip('/')}$", self.entity_configuration))
+        url_map.append(
+            (f"^{self.pre_request_url.lstrip('/')}$", self.pre_request_endpoint))
+        url_map.append(
+            (f"^{self.redirect_url.lstrip('/')}$", self.redirect_endpoint))
+        url_map.append(
+            (f"^{self.request_url.lstrip('/')}$", self.request_endpoint))
+        return url_map
+
+    def entity_configuration(self, context, *args):
+        jwk = JWK()
+
+        data = {
+            "exp": 1649590602,
+            "iat": 1649417862,
+            "iss": "https://rp.example.it",
+            "sub": "https://rp.example.it",
+            "jwks": {
+                "keys": [jwk.export_public()]
+            },
+            "metadata": {
+                "wallet_relying_party": self.config['wallet_relying_party']
+            }
+        }
+
+        jwshelper = JWSHelper(jwk)
+
+        return Response(
+            jwshelper.sign(
+                plain_dict=data,
+                protected={
+                    "alg": "RS256",
+                    "kid": "2HnoFS3YnC9tjiCaivhWLVUJ3AxwGGz_98uRFaqMEEs",
+                    "typ": "entity-statement+jwt"
+                }
+            ),
+            status="200 OK"
+        )
+
+    def pre_request_endpoint(self, context, *args):
+        payload = {'client_id': self.client_id,
+                   'request_uri': self.complete_request_url}
+        query = urlencode(payload, quote_via=quote_plus)
+        response = base64.b64encode(
+            bytes(f'eudiw://authorize?{query}', 'UTF-8'))
+
+        return Response(
+            response,
+            status="200 OK",
+            content="text/json; charset=utf8"
+        )
+
+    def redirect_endpoint(self, context, *args):
+        jwk = JWK()
+
+        helper = JWSHelper(jwk)
+        jwt = helper.sign({
+            "jti": "f47c96a1-f928-4768-aa30-ef32dc78aa69",
```

Comment:
please use a uuid() function here

Reply:
Resolved in [14e8565](https://github.com/italia/eudi-wallet-it-python/pull/14/commits/14e8565f4c4a4faf0d274567db51407bc74f10d4)